workflow/docs/design.md

```diff
@@ -0,0 +1,406 @@
+| :-------------------- | :------------------------------------------------------------------------------------------------------ | :----------------------------------------------------------------------------------------------- | :------------------------------------------------------------------------------------------------------ | :------ | :----- |
+| NodeCreated           | Change phase to `Running`, create CRD about chaos by Actor, notify itself with `NodeChaosInjectSucceed` | -                                                                                                | -                                                                                                       | -       | -      |
+| NodeChaosInjected     | -                                                                                                       | Change phase to `Holding`, then delay notify itself with event `NodeHoldingAwake` and `duration` | -                                                                                                       | -       | -      |
+| NodeHoldingAwake      | -                                                                                                       | -                                                                                                | Change to phase `Running` ,remove CRD about chaos by Actor, notify itself with event `NodeChaosCleaned` | -       | -      |
+| NodeChaosCleaned      | -                                                                                                       | Change phase to `Succeed`, notify parent node(if exists) with `ChildNodeSucceed`                 | -                                                                                                       | -       | -      |
+
+### States for Workflow
+
+There are 4 phase of one Workflow:
+
+- Init
+- Running
+- Succeed
+- Failed
+
+**Init** is the default phase when **Workflow** just created. It means there are no Node is in **Running** or **Holding** state, and it's safe for scheduling next operation.
+
+**Running** means at least 1 node is in **Running**/**Holding**/**WaitingForChild**/**WaitingForSchedule** state.
+
+**Succeed** means all nodes is in **Succeed** node and no Template could schedule anymore.
+
+**Failed** means at least one Node is in **Failed** state.
+
+> Question: Should we interpret other node when Workflow fall into **Failed**?
+> I think we should do that.
+
+### Event
+
+> Unimplemented now.
+
+Inspired by kubernetes events, it should emit an event when something happens:
+
+- chaos or other resource created
+- make decision about conditional branches
+- unexpected errors
+
+### Others
+
+- Persistent Repo or just "repo", fetches and updates status for Workflow and Node.
+
+Detail about instating template:
+
+After calling `Scheduler.ScheduleNext()`, it will return
+
+- which template should be instated
```

Comment:
what is the meaning of `instated`, is it a typo?

Reply:
Yes, it's a typo. I will fix it right now.

Reply:
Fixed! PTAL